test/cpp/jit/test_custom_class_registrations.cpp

```diff
@@ -339,9 +339,10 @@ TORCH_LIBRARY(_TorchScriptTesting, m) {
           },
           [](std::vector<int64_t> state) { // __setstate__
             return c10::make_intrusive<Foo>(state[0], state[1]);
-          });
-  m.def(
-      "takes_foo(__torch__.torch.classes._TorchScriptTesting._Foo foo, Tensor x) -> Tensor");
+          })
+      .def_meta([](c10::intrusive_ptr<Foo> self) { // __get_metadata__
```

Comment:
name bikeshedding:

Btw, if we're going for `__get_metadata__`, we probably want a better name for it. For Tensor Subclasses, the analogous API is named `__tensor_flatten__`.

Reply:
`__obj_flatten__` then?